### ISSUE TITLE
All Options Are Greyed Out

### ISSUE THREAD

**Issue by mrwizzer** [Itaybre/CameraController#81]
This looks promising, but all the options are greyed out for me under both the Basic and Advanced tabs.  (Brightness, Contrast, Zoom, Tilt, etc.)

I'm using a Razer Kiyo as well as the built-in cameras on a 2-year-old iMac Pro running Catalina and a brand new MacBook Pro running Monterey, same results on both.

Thanks for any tips on getting this up and running.

-Mike

**Comment by Itaybre** [comment]
It is normal for built-in cameras to not work, since apple doesn't allow apps to communicate directly with them.

But the razer camera should work fine, this happens on both computers?

**Comment by mrwizzer** [comment]
> It is normal for built-in cameras to not work, since apple doesn't allow apps to communicate directly with them.
> 
> But the razer camera should work fine, this happens on both computers?
> 
> 

I stand corrected, it apparently works fine on the Razer on the iMac Pro, but not on the new MacBook Pro.  Is it because the MacBook Pro has an M1 chip?

Am I out of luck on the MacBook Pro?

Thanks for the prompt response!

-Mike

**Comment by Itaybre** [comment]
That's odd, it should work fine. 

> It is normal for built-in cameras to not work, since apple doesn't allow apps to communicate directly with them. But the razer camera should work fine, this happens on both computers?
> I stand corrected, it apparently works fine on the Razer on the iMac Pro, but not on the new MacBook Pro. Is it because the MacBook Pro has an M1 chip? Am I out of luck on the MacBook Pro? Thanks for the prompt response!
> […](#)

**Comment by DohMoose** [comment]
> It is normal for built-in cameras to not work, since apple doesn't allow apps to communicate directly with them. But the razer camera should work fine, this happens on both computers?
> I stand corrected, it apparently works fine on the Razer on the iMac Pro, but not on the new MacBook Pro. Is it because the MacBook Pro has an M1 chip? Am I out of luck on the MacBook Pro? Thanks for the prompt response!
> […](#)

I'm on an M1 Mac with a Kiyo. I just restarted the app and everything was enabled!

**Comment by strongface99** [comment]
Hi, I am having the greyed out issue, but only on Zoom/Pan/Tilt. This is really what I am hoping to have as my camera has a very wide FOV which doesn't look great. 

I have a Lenovo Essential FHD Webcam, connected to an M1 Mac Mini. 

Thanks

**Comment by dezerving** [comment]
> Hi, I am having the greyed out issue, but only on Zoom/Pan/Tilt. This is really what I am hoping to have as my camera has a very wide FOV which doesn't look great.
> 
> I have a Lenovo Essential FHD Webcam, connected to an M1 Mac Mini.
> 
> Thanks

I was able to get this to work by doing the following (not sure which one triggered it to ungrey everything):
- Gave CameraController "Full Disk Access" permissions
- Went to Control Center and made sure portrait effect wasn't on for the camera
- Restarted the app multiple times

I'm on a M1 MacBook Pro.

**Comment by joekw** [comment]
It's weird, I get all the expected controls for my Kiyo Pro while running the last built release, but when running directly from Xcode, all the controls are greyed out.

It looks like it's failing to create the USBDevice. In AVCaptureDevice.usbDevice(), the interfaceRef is always nil.

**Comment by levyjonlevy** [comment]
Zoom/pan/tilt greyed out for me as well. Laptop is a Macbook Pro 13" 2020 model, running Monterey, webcam is the built-in webcam on a Lenovo ThinkVision T24v-20 monitor.

**Comment by annikolov** [comment]
Zoom/pan/tilt greyed out for me as well. M1 Mac Mini on Ventura 13.0.1 with Vitade 682H Camera.

**Comment by noellered** [comment]
On MacBook Pro (~2019) Intel Chip this happens to me too. I've noticed it more when I have the camera open on another app before opening CameraController (i.e. Zoom, Google Meet)

**Comment by chansearrington** [comment]
Just installed the app and all the settings are greyed out for me as well for both of my Mevo Start cameras over UVC.

I'm also on an M1 Macbook Pro.

About to try setting the app to launch automatically at login and restarting my computer.

**Comment by 1951FDG** [comment]
Same here, this happened when upgrading to macOS Ventura, also auto focus doesn't seem to be working anymore,

Im using a Razer Kiyo X

**Comment by chansearrington** [comment]
> 

A restart fixed it for me.

**Comment by mitkel** [comment]
Had the same issue. Additionally, if I evener managed to "ungray" the controllers, the auto focus option still didn't work for me. Found the following threads: [thread1](https://www.reddit.com/r/razer/comments/glte0s/razer_kiyo_wont_focus/) [thread2](https://www.reddit.com/r/techsupport/comments/gpyc4l/razer_kiyo_webcam_auto_focus_issue_can_anyone_help/) helpful. 

TL;DR - unpluging and pluging in the USB cable did the trick (instead of restarting).

**Comment by 1951FDG** [comment]
I can confirm, unplugging and plugging in the USB cable works! Autofocus working again!

**Comment by 1951FDG** [comment]
Opened CameraController app again today, options not greyed out, closed the app and reopened the app, options greyed out, but settings were still applied to camera.

**Comment by madsenjk** [comment]
> I was able to get this to work by doing the following (not sure which one triggered it to ungrey everything):
> 
> * Gave CameraController "Full Disk Access" permissions
> ...

Thank you for this suggestion. Your first suggestion alone solved the grayed out issue for me. For the first 5 or 6 times I opened the app on first install it had worked fine, allowing full control of my external camera (Dell Pro 2K). Out of nowhere the next 5 times I opened the app there was no way to get the bars recover from greyed out - with no ability to edit. After coming here, and following your suggestion to allow "Full Disk Access" to the application all appears to be available again. 

CameraController 1.4.0 (6) - homebrew install
Ventura 13.2.1
Dell Pro 2K WB5023

**Comment by quicksite** [comment]
Me too...  But only now have I read dev's comment that

> It is normal for built-in cameras to not work, since apple doesn't allow apps to communicate directly with them.
https://github.com/Itaybre/CameraController/issues/81#issuecomment-1275368397

Too bad. I did not know this.

**Comment by martonpaulo** [comment]
Same issue here. All settings are grayed out.

I’ve tried using both the built-in Mac webcam and iPhone mirroring, but neither worked.

I also granted full disk access to CameraController, restarted the app, iPhone, and Mac, but it didn’t resolve the issue.

Device Details:
MacBook Air M1
Sequoia 15.1.1

**Comment by Itaybre** [comment]
> Same issue here. All settings are grayed out.
> 
> I’ve tried using both the built-in Mac webcam and iPhone mirroring, but neither worked.
> 
> I also granted full disk access to CameraController, restarted the app, iPhone, and Mac, but it didn’t resolve the issue.
> 
> Device Details: MacBook Air M1 Sequoia 15.1.1

You cannot control this settings in either the built in webcam or iPhone mirroring.
CameraController only works with USB cameras

**Comment by hjjl26** [comment]
So MacBook Pro with M1 Pro running Sonoma 14.7.7 everything greyed out with USB camera Yongnuo YN 433 get image but no control, screen shot attached. Any thoughts? tried most suggestions up above. Thanks!

<img width="1035" height="966" alt="Image" src="https://github.com/user-attachments/assets/85a1e52e-7553-4802-bb54-1896b554b192" />
